api/server/services/Config/__tests__/getCachedTools.spec.js

```diff
@@ -0,0 +1,408 @@
+const { CacheKeys } = require('librechat-data-provider');
+const getLogStores = require('~/cache/getLogStores');
+
+jest.mock('~/cache/getLogStores');
+
+const {
+  ToolCacheKeys,
+  getCachedTools,
+  setCachedTools,
+  getMCPServerTools,
+  invalidateCachedTools,
+} = require('../getCachedTools');
+
+describe('getCachedTools - Cache Isolation Security', () => {
+  let mockCache;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockCache = {
+      get: jest.fn(),
+      set: jest.fn(),
+      delete: jest.fn(),
+    };
+
+    getLogStores.mockReturnValue(mockCache);
+  });
+
+  describe('ToolCacheKeys.MCP_SERVER', () => {
+    it('should generate cache keys that include userId', () => {
+      const key = ToolCacheKeys.MCP_SERVER('user123', 'github');
+      expect(key).toBe('tools:mcp:user123:github');
+    });
+
+    it('should generate different cache keys for different users with same server', () => {
+      const keyUserA = ToolCacheKeys.MCP_SERVER('userA', 'github');
+      const keyUserB = ToolCacheKeys.MCP_SERVER('userB', 'github');
+
+      expect(keyUserA).not.toEqual(keyUserB);
+      expect(keyUserA).toBe('tools:mcp:userA:github');
+      expect(keyUserB).toBe('tools:mcp:userB:github');
+    });
+
+    it('should generate different cache keys for same user with different servers', () => {
+      const keyGithub = ToolCacheKeys.MCP_SERVER('user123', 'github');
+      const keyGitlab = ToolCacheKeys.MCP_SERVER('user123', 'gitlab');
+
+      expect(keyGithub).not.toEqual(keyGitlab);
+      expect(keyGithub).toBe('tools:mcp:user123:github');
+      expect(keyGitlab).toBe('tools:mcp:user123:gitlab');
+    });
+
+    it('should generate unique cache keys for each user-server combination', () => {
+      const combinations = [
+        ['user1', 'server1'],
+        ['user1', 'server2'],
+        ['user2', 'server1'],
+        ['user2', 'server2'],
+      ];
+
+      const keys = combinations.map(([userId, serverName]) =>
+        ToolCacheKeys.MCP_SERVER(userId, serverName),
+      );
+
+      // All keys should be unique
+      const uniqueKeys = new Set(keys);
+      expect(uniqueKeys.size).toBe(keys.length);
+    });
+  });
```

Comment:
These tests are overkill. They are all testing that with different input params for `ToolCacheKeys.MCP_SERVER`, different results are returned and they conform to a specific format. Since the test before these already shows us that the key is constructed from the input params in the `tools:mcp:<userId>:<serverName>` format, testing that having different combinations of userId/serverName resulting different keys is quite redundant. At most we only need 1 of such tests.